xilem/examples/virtual_cats.rs

```diff
@@ -71,50 +75,74 @@ impl VirtualCats {
-                    } else {
+                move |state: &mut Self, message| {
+                    let Some(status) = state.statuses.iter_mut().find(|it| it.code == code) else {
                         unreachable!("We never remove items from `statuses`")
+                    };
+                    match message {
+                        Ok(image) => {
+                            status.image = ImageState::Available(image);
+                        }
+                        Err(err) => status.image = ImageState::Error(err),
                     }
                 },
             ))
         } else {
             None
         };
-        let img = if let Some(img) = img {
-            let attribution = sized_box(
-                sized_box(
-                    prose("Copyright ©️ https://http.cat")
-                        .line_break_mode(LineBreaking::Clip)
-                        .text_alignment(TextAlign::End),
+        let img = match &item.image {
+            ImageState::Available(img) => {
+                let attribution = sized_box(
+                    sized_box(
+                        prose("Copyright ©️ https://http.cat")
+                            .line_break_mode(LineBreaking::Clip)
+                            .text_alignment(TextAlign::End),
+                    )
+                    .padding(4.)
+                    .corner_radius(4.)
+                    .background_color(palette::css::BLACK.multiply_alpha(0.5)),
                 )
-                .padding(4.)
-                .corner_radius(4.)
-                .background_color(palette::css::BLACK.multiply_alpha(0.5)),
-            )
-            .padding(Padding {
-                left: 0.,
-                right: 42.,
-                top: 30.,
-                bottom: 0.,
-            });
-            Either::A(zstack((
-                image(img).fit(ObjectFit::FitWidth),
-                attribution.alignment(UnitPoint::TOP_RIGHT),
-            )))
-        } else {
-            Either::B(sized_box(spinner()).width(80.px()).height(80.px()))
+                .padding(Padding {
+                    left: 0.,
+                    right: 42.,
+                    top: 30.,
+                    bottom: 0.,
+                });
+                let imgview = zstack((
+                    image(img).fit(ObjectFit::FitWidth),
+                    attribution.alignment(UnitPoint::TOP_RIGHT),
+                ));
+                OneOf3::A(imgview)
+            }
+            ImageState::Pending => OneOf3::B(sized_box(spinner()).width(80.px()).height(80.px())),
+            ImageState::Error(err) => {
+                // the people deserve their cat.
+                // It is vital that the cat explains what went wrong.
+                let asciicat = label(
+                    "
+                  /\\_/\\  !
+                 ( o.o )
+                  > ^ <",
+                )
+                .color(BLACK);
```

Comment:
I'm of two minds here. This is of course cute, which is good.
However, I can see a few issues:
- Is there attribution for this art? We need to make sure we're following whatever license it's under. That adds complication.
- For accessibility purposes, I suspect that this is pretty hard to understand. We don't have good ways to exclude things from accessibility currently.

Maybe we can just use a 😿 instead?